File: sample/src/main/java/me/panpf/adapter/sample/adapter/AssemblyStickyRecyclerAdapter.java
package me.panpf.adapter.sample.adapter;

import java.util.List;

import me.panpf.adapter.AssemblyRecyclerAdapter;
import me.panpf.adapter.ItemFactory;
import me.panpf.recycler.sticky.StickyRecyclerAdapter;

public class AssemblyStickyRecyclerAdapter extends AssemblyRecyclerAdapter implements StickyRecyclerAdapter {
    @Override
    public boolean isStickyItemByType(int type) {
        List<ItemFactory> itemFactoryList = getItemFactoryList();
        if (itemFactoryList != null && itemFactoryList.size() > 0) {
            for (ItemFactory itemFactory : itemFactoryList) {
                if (itemFactory.getItemType() == type && itemFactory instanceof StickyItemFactory) {
                    return true;
                }
            }
        }
        return false;
    }

    public interface StickyItemFactory {

    }
}

File: sample/src/main/java/me/panpf/adapter/sample/adapter/AssemblyStickyRecyclerAdapter.kt
package me.panpf.adapter.sample.adapter

import me.panpf.adapter.AssemblyRecyclerAdapter
import me.panpf.recycler.sticky.StickyRecyclerAdapter

class AssemblyStickyRecyclerAdapter : AssemblyRecyclerAdapter(), StickyRecyclerAdapter {
    override fun isStickyItemByType(type: Int): Boolean {
        return getItemFactoryByViewType(type) is StickyItemFactory
    }

    interface StickyItemFactory
}